src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_4_5/VehiclePassengers.java

```diff
@@ -24,10 +26,20 @@
 	public boolean postFromServerRead() {
 		if (passengersIds.length == 0) {
 			passengerId = vehiclePassenger.remove(vehicleId);
+			passengerToVehicle.remove(passengerId);
 		} else {
 			passengerId = passengersIds[0];
 			vehiclePassenger.put(vehicleId, passengerId);
 		}
+		WatchedEntityCache entityCache = cache.getWatchedEntityCache();
+		if (passengersIds.length != 0 && entityCache.hasWatchedEntity(vehicleId) && entityCache.hasWatchedEntity(passengerId)) {
+			int prevVehicleId = passengerToVehicle.getOrDefault(passengerId, -1);
+			if (entityCache.hasWatchedEntity(prevVehicleId)) {
+				return false;
+			} else {
+				passengerToVehicle.put(passengerId, vehicleId);
+			}
+		}
```

Comment:
This for problem that if non-passenger rejoin, minecart passenger looks flying above 2~3 blocks.
Cause 1.12.2 bukkit sends PacketPlayOutMount(VehiclePassenger) 3times. 1.5.2 isn't